module/VuFindSearch/src/VuFindSearch/Backend/Feature/ConnectorCacheTrait.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Caching support trait for connectors.
+ *
+ * Prerequisites:
+ *
+ * - LoggerAwareInterface
```

Comment:
The methods you depend on here come from LoggerAwareTrait, but are not required by LoggerAwareInterface. Maybe it would be better to just compose the trait here instead of referencing it via a comment.

Reply:
I changed that to LoggerAwareTrait. I'm not really a fan of using traits in traits since it's confusing when looking at the main class that doesn't directly use a trait but accesses its methods. Also, didn't it cause a problem when a class uses two traits that in turn use the same trait (thinking about the OAuth2 trait with ILS drivers)?

Reply:
Yes, that makes sense. There's just no good way to safely compose traits, which is frustrating. (Possibly a sign that we're pushing this mechanism too hard... but it is very useful when it works).

A few alternatives that might be worth considering:

Option 1: Revert to a dependency on LoggerAwareInterface, but instead of using the LoggerAwareTrait's convenience methods, instead just call methods directly on `$this->logger` inside an `if ($this->logger ?? false) {` block. That way, failing to compose the dependency isn't a fatal problem. You could even put in an else with an `error_log` message about a missing logger.

Option 2: Keep the dependency on LoggerAwareTrait but add `is_callable` checks.

Option 3: Create a private helper message for logging within the trait, and do Option 1 or Option 2 inside it.

Maybe I'm overthinking all of this, but I just hate the idea that somebody can use a trait and create a situation where a fatal error triggers on the basis of a rare condition, and when it occurs, it would be rather difficult to figure out why.

Reply:
No overthinking, it's good to do this properly. How about now?

Reply:
I decided to check for logger with the correct interface so that it doesn't need to be any of the "...Aware..." things, but logger itself needs to implement the correct interface.

Reply:
Also, I think a shared base class for connectors would help here structure the code better.

Reply:
Thanks, this looks good to me. I'll do a bit of testing later today (though I can only test EDS) and report back if I encounter any problems.

The idea of a common connector base class is a good idea -- do you think it's worth a JIRA ticket to capture for future reference? I don't think it's our top priority right now, but definitely a "nice to have" for the future. The trick is that not every backend even necessarily has to have a separate connector class, and some backends rely on third-party libraries as connectors, so I don't think we want to enforce any specific interface across all backends... but that doesn't mean we can't have a baseline for internal connectors as an optional best practice.

Reply:
Yes. I added [a ticket](https://vufind.org/jira/browse/VUFIND-1549).

Reply:
Thanks! I added a comment there to summarize this thread (and link back to this PR for context).